autopilot/betweenness_centrality.go

```diff
@@ -150,10 +162,48 @@ func (bc *BetweennessCentrality) Refresh(graph ChannelGraph) error {
 		return err
 	}
 
-	// TODO: parallelize updates to centrality.
-	centrality := make([]float64, len(cache.Nodes))
+	var wg sync.WaitGroup
+	work := make(chan int)
+	partials := make(chan []float64, bc.workers)
+
+	// Each worker will compute a partial result.
+	worker := func() {
+		defer wg.Done()
+		partial := make([]float64, len(cache.Nodes))
+		for {
+			node, ok := <-work
+			if !ok {
+				break
+			}
+			betweennessCentrality(cache, node, partial)
```

Comment:
IMO it is a bit confusing that the result of the calculation is added by reference to the partial slice, instead of returned. 

Now that we are collecting and summing the final results below anyway, what do you think about letting each worker return the result for each calculation immediately, then collect them all below? So it would be something like

```golang
work := make(chan int, len(cache.Nodes))
for node := range cahce.Nodes {
    work <- node
}
close(work)

worker := func() {
     defer wg.Done()
      
     for node := range work {
             partials <- betweennessCentrality(cache, node)
      }
}

wg.Add(bc.workers)
for i := range bc.workers {
    go worker()
}

wg.Wait()
...
```

Not sure if this would have a performance impact though, sending on the channe after every node calculation, but it should be unblocking

Reply:
With the current approach we allocate NumCPU() * N buffers and create partials without sending the result on a channel, resulting in less contention/blocking on channel communication. Also in the end we do much less operations with the current approach as the one above would require summing N * N numbers vs N * NumCPU().

Reply:
discussed offline. The current approach is better since it allocates less memory for the partial results.